src/libs/nav2_tasks/include/nav2_tasks/task_client.hpp

```diff
@@ -70,6 +70,24 @@ class TaskClient
     cancelPub_->publish(msg);
   }
 
+  bool waitForServer(std::chrono::milliseconds timeout = std::chrono::milliseconds(3000))
+  {
+    std::string taskName = getTaskName<CommandMsg, ResultMsg>();
+    taskName += "_command";
+
+    auto startTime = std::chrono::high_resolution_clock::now();
+
+    while (node_->count_subscribers(taskName) < 1) {
+      rclcpp::spin_some(node_->get_node_base_interface());
```

Comment:
We're going to consume 100% CPU while waiting here. Some sort of sleep is probably appropriate before the next try.